packages/tableofcontents.lua

```diff
@@ -30,6 +30,7 @@ SILE.registerCommand("tableofcontents", function (options, content)
   local tocfile,_ = io.open(SILE.masterFilename .. '.toc')
   if not tocfile then
     SILE.call("tableofcontents:notocmessage")
+	SILE.passes = SILE.passes + 1
```

Comment:
This is a wrong approach, IMHO, with a real bad separation of concerns if a package (here `tableofcontents`) increases a global counter.
Other packages could have a similar need of "auxiliary files" generated along the way to be updated by multiple passes (e.g. cross-references; lists of figures or tables if they decide not to leverage the ToC system; indexes; alternate not even mentioning alternate implementation of the ToC would allow multiple ToC subsets or ToC per chapters, etc.).

The general case cannot be solved this way. These packages are not all going to do a `+ 1`... They cannot know.

Reply:
You're right, this global usage was a wrong approach. Packages could set some global status flag for requesting a rebuild, but they shouldn't be in charge of marking the current pass status.